[CSS Zoom] Fix font-relative units for unzoomed properties

vitorroriz opened the pull request:
#### 4082642c7f1c6335700c6e1a64e362ce92d13758
<pre>
[CSS Zoom] Fix font-relative units for unzoomed properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=299816">https://bugs.webkit.org/show_bug.cgi?id=299816</a>
<a href="https://rdar.apple.com/161590209">rdar://161590209</a>

Reviewed by Brent Fulgham.

In our current architecture, the computed font size has already zoom applied
to it.

Properties that are tagged as “unzoomed” (assuming evaluation flag is enabled)
will have zoom applied to its value during evaluation. If such a value uses a
font-related unit, we would have double zoom applied. So, for such cases we want
to use the unzoomed computed size (which is basically computedSize / zoom, not
necessarily equal to the specified size).

If we indiscriminately use the unzoomed computed font-size when resolving length
this would break the the resolution for all non-converted “length” properties (since
evaluate won’t apply zoom for them during style resolution).

Therefore, we need to request the use of the “unzoomed computed font-size” depending on
the type we are resolving. In order to do that, we propagating RangeZoomOptions via
CSSToLengthConversionData. In that way, we can request the use of the unzoomed computed
font size just when the EvaluationTimeZoomEnabled flag is enabled and when RangeZoomOptions
is RangeZoomOptions::Unzoomed.

Each new unzoomed property will need to set RangeZoomOptions::Unzoomed at
CSSToLengthConversionData in order to font relative units to get evaluated correctly
from now on.

There are different ways this can be done. Here are some exaples:

1. Via copyWithAdjustedZoom: StyleWordSpacing, uses copyWithAdjustedZoom for mutating the
original state&apos;s CSSToLengthConversionData. This funciton can now receive a RangeZoomOptions:
```
state.cssToLengthConversionData().copyWithAdjustedZoom(zoom, CSS::RangeZoomOptions::Unzoomed)
```

2. Via its own copy method. StyleLineHeight already has its own method to produce a conversion data,
in that case it makes sense to explicitly set it there:
```
    CSSToLengthConversionData copyForLineHeight(float zoom) const
    {
        CSSToLengthConversionData copy(*this);
        copy.m_zoom = zoom;
        copy.m_propertyToCompute = CSSPropertyLineHeight;
        copy.m_rangeZoomOption = CSS::RangeZoomOptions::Unzoomed;
        return copy;
    }
```

3. If your property uses toStyleFromCSSValue&lt;LineWidth::Length&gt;(state, value), you might
notice that there are no conversionData propagation. Here, the conversion data is created
by selectConversionData or by the conversion operator CSSToLengthConversionData operator().
For those, this patch are already propagates CSS::RangeZoomOptions::Unzoomed taking advantage
of the existent branches for zoom options there:
Example:
```
    selectConversionData(BuilderState&amp; builderState)
    {
        ...
        else if constexpr (R.zoomOptions == CSS::RangeZoomOptions::Unzoomed) {
        if (shouldUseEvaluationTimeZoom(builderState))
            return builderState.cssToLengthConversionData().copyWithAdjustedZoom(1.0f, CSS::RangeZoomOptions::Unzoomed);
        ...
    }
```

Note: Ideally we would have a function template that given a type could return what type of
computed font-size should be used, given that this depends on the StrongTypes which are
already marked with RangeZoomOptions::Unzoomed. This would require some complicate generic
templates given that we have to unroll differente lenght wrappers like, for example:

struct WordSpacing : LengthWrapperBase&lt;LengthPercentage&lt;CSS::AllUnzoomed&gt;&gt;
Length = Style::Length&lt;CSS::NonnegativeUnzoomed&gt;;

Until we do that, I&apos;m propagating it dynamically here.

* LayoutTests/fast/css/line-height-zoom-get-computed-style.html:
* LayoutTests/fast/sub-pixel/zoomed-em-border.html:
* LayoutTests/fast/text/line-height-minimumFontSize-zoom.html:
* LayoutTests/svg/zoom/page/zoom-replaced-intrinsic-ratio-001.htm:
* Source/WebCore/css/CSSToLengthConversionData.h:
(WebCore::CSSToLengthConversionData::rangeZoomOption const):
(WebCore::CSSToLengthConversionData::copyWithAdjustedZoom const):
(WebCore::CSSToLengthConversionData::copyForLineHeight const):
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericRange.h:
* Source/WebCore/platform/graphics/FontCascadeDescription.cpp:
* Source/WebCore/platform/graphics/FontDescription.cpp:
(WebCore::m_enableEvaluationTimeZoom):
(WebCore::m_shouldDisableLigaturesForSpacing): Deleted.
* Source/WebCore/platform/graphics/FontDescription.h:
(WebCore::FontDescription::usedZoomFactor const):
(WebCore::FontDescription::computedSizeForRangeZoomOption const):
(WebCore::FontDescription::unzoomedComputedSize const):
(WebCore::FontDescription::enableEvaluationTimeZoom const):
(WebCore::FontDescription::setComputedSize):
(WebCore::FontDescription::setEnableEvaluationTimeZoom):
* Source/WebCore/style/StyleBuilderState.cpp:
(WebCore::Style::BuilderState::updateFontForSizeChange):
(WebCore::Style::BuilderState::setFontSize):
* Source/WebCore/style/StyleBuilderStateInlines.h:
(WebCore::Style::BuilderState::setFontDescriptionFontSize):
* Source/WebCore/style/StyleFontSizeFunctions.cpp:
(WebCore::Style::computedFontSizeFromSpecifiedSize):
* Source/WebCore/style/StyleFontSizeFunctions.h:
* Source/WebCore/style/StyleResolveForDocument.cpp:
(WebCore::Style::resolveForDocument):
* Source/WebCore/style/StyleResolveForFont.cpp:
(WebCore::Style::fontSizeFromUnresolvedFontSize):
* Source/WebCore/style/StyleResolver.cpp:
(WebCore::Style::Resolver::defaultStyleForElement):
* Source/WebCore/style/values/inline/StyleLineHeight.cpp:
(WebCore::Style::CSSValueConversion&lt;LineHeight&gt;::operator):
* Source/WebCore/style/values/primitives/StyleLengthResolution.cpp:
(WebCore::Style::computeUnzoomedNonCalcLengthDouble):
(WebCore::Style::computeNonCalcLengthDouble):
* Source/WebCore/style/values/primitives/StyleLengthResolution.h:
* Source/WebCore/style/values/primitives/StyleLengthWrapper+CSSValueConversion.h:
(WebCore::Style::CSSValueConversion&lt;T&gt;::selectConversionData):
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+CSSValueConversion.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Conversions.h:
* Source/WebCore/style/values/text/StyleWordSpacing.cpp:
(WebCore::Style::CSSValueConversion&lt;WordSpacing&gt;::operator):

Canonical link: <a href="https://commits.webkit.org/301757@main">https://commits.webkit.org/301757@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d21a9edc3674ef246ef552c2971d2fdc51ec78c7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/126958 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/46594 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/37584 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/133962 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/78533 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/128829 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/47211 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/55122 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/96608 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/78533 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/129906 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/37778 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/113559 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/77120 "Passed tests") | | ⏳ 🛠 vision-apple 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/36640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/31725 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/77356 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/107615 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/32046 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/136487 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/53614 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/41278 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/105126 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/54117 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/109923 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/104817 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26728 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/50333 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/28674 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/51093 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/53546 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/59418 "Built successfully") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/125/builds/52786 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/129/builds/56120 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/54546 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mdubet commented on Source/WebCore/platform/graphics/FontDescription.cpp:
```diff
@@ -68,6 +68,7 @@ FontDescription::FontDescription()
     , m_fontStyleAxis(enumToUnderlyingType(FontStyleAxis::slnt))
     , m_shouldAllowUserInstalledFonts(enumToUnderlyingType(AllowUserInstalledFonts::No))
     , m_shouldDisableLigaturesForSpacing(false)
+    , m_enableEvaluationTimeZoom(false)
```
`enable` in a boolean is weird (it's an action). We could just use `bool m_evaluationTimeZoom` and the setter would be `setEvaluationTimeZoom(true/false)`

vitorroriz replied:
I'm choosing the same name as we use in RenderStyle. I guess we can rename everything later for clarity.

brentfulgham replied:
Agreed. This is a good idea for a follow-up cleanup, but not needed right now.

mdubet commented on Source/WebCore/style/StyleFontSizeFunctions.h:
```diff
@@ -36,9 +36,13 @@ struct SettingsValues;
 enum class MinimumFontSizeRule : uint8_t { None, Absolute, AbsoluteAndRelative };
+struct ComputedFontSize {
+    float size;
+    float usedZoomFactor;
```
Should this have default values?

brentfulgham replied:
Seems like defaults would be reasonable to include here!

nullhook commented on Source/WebCore/style/values/primitives/StyleLengthWrapper+CSSValueConversion.h:
```diff
@@ -72,7 +72,7 @@ template<LengthWrapperBaseDerived T> struct CSSValueConversion<T> {
         } else if constexpr (T::Fixed::range.zoomOptions == CSS::RangeZoomOptions::Unzoomed) {
             if (shouldUseEvaluationTimeZoom(builderState))
-                return builderState.cssToLengthConversionData().copyWithAdjustedZoom(1.0f);
+                return builderState.cssToLengthConversionData().copyWithAdjustedZoom(1.0f, CSS::RangeZoomOptions::Unzoomed);
```
we know that this space is for ::Unzoomed. rather than hardcoding here can we pass `T::Fixed::range.zoomOptions` ?

vitorroriz commented on Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+CSSValueConversion.h:
```diff
@@ -124,6 +124,7 @@ template<auto R, typename V> struct CSSValueConversion<Length<R, V>> {
         if (!protectedValue)
             return 0_css_px;
 
+        // Vitor FIXME: we might be able to identify type here and set our flag to conversion data
```
oops, I'll remove that once EWS finishes running

nullhook commented on Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+CSSValueConversion.h:
```diff
@@ -105,10 +105,10 @@ template<auto R, typename V> struct CSSValueConversion<Length<R, V>> {
         } else if constexpr (R.zoomOptions == CSS::RangeZoomOptions::Unzoomed) {
             if (shouldUseEvaluationTimeZoom(builderState))
-                return builderState.cssToLengthConversionData().copyWithAdjustedZoom(1.0f);
+                return builderState.cssToLengthConversionData().copyWithAdjustedZoom(1.0f, CSS::RangeZoomOptions::Unzoomed);
```
you can pass `R.zoomOptions`

nullhook commented on Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Conversions.h:
```diff
@@ -60,7 +60,7 @@ template<auto R, typename V> struct ConversionDataSpecializer<CSS::LengthRaw<R,
         } else if constexpr (R.zoomOptions == CSS::RangeZoomOptions::Unzoomed) {
             if (shouldUseEvaluationTimeZoom(state))
-                return state.cssToLengthConversionData().copyWithAdjustedZoom(1.0f);
+                return state.cssToLengthConversionData().copyWithAdjustedZoom(1.0f, CSS::RangeZoomOptions::Unzoomed);
```
ditto

nullhook commented on Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+CSSValueConversion.h:
```diff
@@ -105,10 +105,10 @@ template<auto R, typename V> struct CSSValueConversion<Length<R, V>> {
 
             return builderState.useSVGZoomRulesForLength()
-                ? builderState.cssToLengthConversionData().copyWithAdjustedZoom(1.0f)
+                ? builderState.cssToLengthConversionData().copyWithAdjustedZoom(1.0f, CSS::RangeZoomOptions::Unzoomed)
```
ditto

nullhook commented on Source/WebCore/style/values/text/StyleWordSpacing.cpp:
```diff
@@ -40,7 +40,7 @@ auto CSSValueConversion<WordSpacing>::operator()(BuilderState& state, const CSSV
         if (zoom == state.cssToLengthConversionData().zoom())
             return state.cssToLengthConversionData();
-        return state.cssToLengthConversionData().copyWithAdjustedZoom(zoom);
+        return state.cssToLengthConversionData().copyWithAdjustedZoom(zoom, CSS::RangeZoomOptions::Unzoomed);
```
you can access WordSpacing's range like this `WordSpacing::Fixed::range.zoomOptions`

brentfulgham commented on Source/WebCore/style/StyleResolveForDocument.cpp:
```diff
@@ -87,12 +87,15 @@ RenderStyle resolveForDocument(const Document& document)
         fontDescription.setSpecifiedLocale(document.contentLanguage());
         fontDescription.setOneFamily(standardFamily);
         fontDescription.setShouldAllowUserInstalledFonts(settings.shouldAllowUserInstalledFonts() ? AllowUserInstalledFonts::Yes : AllowUserInstalledFonts::No);
+        // FIXME: We need enableEvaluationTimeZoom to be accessible from FontDescription, not only from RenderStyle. Would it be weird to move it to FontDescription (which is already accessible from RenderStyle)?
```
You should start a thread about this in the CSS Zoom channel to discuss.

brentfulgham reviewed (approved):
This seems to work properly, and I don't see any obvious problems. r=me, but I wish we could get a review from @sammygill, @mdubet, or @anttijk to make sure the style changes are ok.

nullhook commented on Source/WebCore/platform/graphics/FontDescription.h:
```diff
@@ -46,6 +47,9 @@ class FontDescription {
     float computedSize() const { return m_computedSize; }
+    float usedZoomFactor() const { return m_usedZoomFactor; }
+    float computedSizeForRangeZoomOption(CSS::RangeZoomOptions option) const { return (enableEvaluationTimeZoom() && option == CSS::RangeZoomOptions::Unzoomed) ? unzoomedComputedSize() : computedSize(); }
+    float unzoomedComputedSize() const { return m_computedSize / m_usedZoomFactor; }
```
mega nit: move this above `computedSizeForRangeZoomOption` so when reading you can see all the dependencies

nullhook commented on Source/WebCore/style/values/inline/StyleLineHeight.cpp:
```diff
@@ -78,8 +78,9 @@ auto CSSValueConversion<LineHeight>::operator()(BuilderState& state, const CSSPr
+        auto textZoom = shouldUseEvaluationTimeZoom(state) ? conversionData.zoom() : 1.0f;
         return LineHeight::Fixed {
-            CSS::clampToRange<LineHeight::Fixed::range, float>((state.style().computedFontSize() * primitiveValue.resolveAsPercentage<int>(conversionData)) / 100.0, minValueForCssLength, maxValueForCssLength)
+            CSS::clampToRange<LineHeight::Fixed::range, float>((state.style().fontDescription().computedSizeForRangeZoomOption(conversionData.rangeZoomOption()) * primitiveValue.resolveAsPercentage<int>(conversionData) * textZoom) / 100.0, minValueForCssLength, maxValueForCssLength)
```
is there a specific reason we pass `conversionData.rangeZoomOption()` here? rather than using type's intrinsic zoom option? `LineHeight::Fixed::range.zoomOptions`

nullhook commented on Source/WebCore/style/StyleResolveForFont.cpp:
```diff
@@ -271,7 +271,7 @@ static ResolvedFontSize fontSizeFromUnresolvedFontSize(const CSSPropertyParserHe
                             RefPtr document = dynamicDowncast<Document>(context);
                             return {
-                                .size = static_cast<float>(Style::computeUnzoomedNonCalcLengthDouble(lengthPercentage.value, lengthUnit, CSSPropertyFontSize, &fontCascade, document ? document->renderView() : nullptr)),
+                                .size = static_cast<float>(Style::computeUnzoomedNonCalcLengthDouble(lengthPercentage.value, lengthUnit, CSSPropertyFontSize, &fontCascade, CSS::RangeZoomOptions::Default, document ? document->renderView() : nullptr)),
```
you can pass `lengthPercentage.range.zoomOptions`

vitorroriz committed:
[CSS Zoom] Fix font-relative units for unzoomed properties
https://bugs.webkit.org/show_bug.cgi?id=299816
rdar://161590209

Reviewed by Brent Fulgham.

In our current architecture, the computed font size has already zoom applied
to it.

Properties that are tagged as “unzoomed” (assuming evaluation flag is enabled)
will have zoom applied to its value during evaluation. If such a value uses a
font-related unit, we would have double zoom applied. So, for such cases we want
to use the unzoomed computed size (which is basically computedSize / zoom, not
necessarily equal to the specified size).

If we indiscriminately use the unzoomed computed font-size when resolving length
this would break the the resolution for all non-converted “length” properties (since
evaluate won’t apply zoom for them during style resolution).

Therefore, we need to request the use of the “unzoomed computed font-size” depending on
the type we are resolving. In order to do that, we propagating RangeZoomOptions via
CSSToLengthConversionData. In that way, we can request the use of the unzoomed computed
font size just when the EvaluationTimeZoomEnabled flag is enabled and when RangeZoomOptions
is RangeZoomOptions::Unzoomed.

Each new unzoomed property will need to set RangeZoomOptions::Unzoomed at
CSSToLengthConversionData in order to font relative units to get evaluated correctly
from now on.

There are different ways this can be done. Here are some exaples:

1. Via copyWithAdjustedZoom: StyleWordSpacing, uses copyWithAdjustedZoom for mutating the
original state's CSSToLengthConversionData. This funciton can now receive a RangeZoomOptions:
```
state.cssToLengthConversionData().copyWithAdjustedZoom(zoom, CSS::RangeZoomOptions::Unzoomed)
```

2. Via its own copy method. StyleLineHeight already has its own method to produce a conversion data,
in that case it makes sense to explicitly set it there:
```
    CSSToLengthConversionData copyForLineHeight(float zoom) const
    {
        CSSToLengthConversionData copy(*this);
        copy.m_zoom = zoom;
        copy.m_propertyToCompute = CSSPropertyLineHeight;
        copy.m_rangeZoomOption = CSS::RangeZoomOptions::Unzoomed;
        return copy;
    }
```

3. If your property uses toStyleFromCSSValue<LineWidth::Length>(state, value), you might
notice that there are no conversionData propagation. Here, the conversion data is created
by selectConversionData or by the conversion operator CSSToLengthConversionData operator().
For those, this patch are already propagates CSS::RangeZoomOptions::Unzoomed taking advantage
of the existent branches for zoom options there:
Example:
```
    selectConversionData(BuilderState& builderState)
    {
        ...
        else if constexpr (R.zoomOptions == CSS::RangeZoomOptions::Unzoomed) {
        if (shouldUseEvaluationTimeZoom(builderState))
            return builderState.cssToLengthConversionData().copyWithAdjustedZoom(1.0f, CSS::RangeZoomOptions::Unzoomed);
        ...
    }
```

Note: Ideally we would have a function template that given a type could return what type of
computed font-size should be used, given that this depends on the StrongTypes which are
already marked with RangeZoomOptions::Unzoomed. This would require some complicate generic
templates given that we have to unroll differente lenght wrappers like, for example:

struct WordSpacing : LengthWrapperBase<LengthPercentage<CSS::AllUnzoomed>>
Length = Style::Length<CSS::NonnegativeUnzoomed>;

Until we do that, I'm propagating it dynamically here.

* LayoutTests/fast/css/line-height-zoom-get-computed-style.html:
* LayoutTests/fast/sub-pixel/zoomed-em-border.html:
* LayoutTests/fast/text/line-height-minimumFontSize-zoom.html:
* LayoutTests/svg/zoom/page/zoom-replaced-intrinsic-ratio-001.htm:
* Source/WebCore/css/CSSToLengthConversionData.h:
(WebCore::CSSToLengthConversionData::rangeZoomOption const):
(WebCore::CSSToLengthConversionData::copyWithAdjustedZoom const):
(WebCore::CSSToLengthConversionData::copyForLineHeight const):
* Source/WebCore/css/values/primitives/CSSPrimitiveNumericRange.h:
* Source/WebCore/platform/graphics/FontCascadeDescription.cpp:
* Source/WebCore/platform/graphics/FontDescription.cpp:
(WebCore::m_enableEvaluationTimeZoom):
(WebCore::m_shouldDisableLigaturesForSpacing): Deleted.
* Source/WebCore/platform/graphics/FontDescription.h:
(WebCore::FontDescription::usedZoomFactor const):
(WebCore::FontDescription::computedSizeForRangeZoomOption const):
(WebCore::FontDescription::unzoomedComputedSize const):
(WebCore::FontDescription::enableEvaluationTimeZoom const):
(WebCore::FontDescription::setComputedSize):
(WebCore::FontDescription::setEnableEvaluationTimeZoom):
* Source/WebCore/style/StyleBuilderState.cpp:
(WebCore::Style::BuilderState::updateFontForSizeChange):
(WebCore::Style::BuilderState::setFontSize):
* Source/WebCore/style/StyleBuilderStateInlines.h:
(WebCore::Style::BuilderState::setFontDescriptionFontSize):
* Source/WebCore/style/StyleFontSizeFunctions.cpp:
(WebCore::Style::computedFontSizeFromSpecifiedSize):
* Source/WebCore/style/StyleFontSizeFunctions.h:
* Source/WebCore/style/StyleResolveForDocument.cpp:
(WebCore::Style::resolveForDocument):
* Source/WebCore/style/StyleResolveForFont.cpp:
(WebCore::Style::fontSizeFromUnresolvedFontSize):
* Source/WebCore/style/StyleResolver.cpp:
(WebCore::Style::Resolver::defaultStyleForElement):
* Source/WebCore/style/values/inline/StyleLineHeight.cpp:
(WebCore::Style::CSSValueConversion<LineHeight>::operator):
* Source/WebCore/style/values/primitives/StyleLengthResolution.cpp:
(WebCore::Style::computeUnzoomedNonCalcLengthDouble):
(WebCore::Style::computeNonCalcLengthDouble):
* Source/WebCore/style/values/primitives/StyleLengthResolution.h:
* Source/WebCore/style/values/primitives/StyleLengthWrapper+CSSValueConversion.h:
(WebCore::Style::CSSValueConversion<T>::selectConversionData):
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+CSSValueConversion.h:
* Source/WebCore/style/values/primitives/StylePrimitiveNumericTypes+Conversions.h:
* Source/WebCore/style/values/text/StyleWordSpacing.cpp:
(WebCore::Style::CSSValueConversion<WordSpacing>::operator):

Canonical link: https://commits.webkit.org/301757@main
